File: flutter/lib/models/association.dart
class Association {
  final String? id;
  final String name;
  final String phone;
  final String email;
  final String address;
  final String number;
  final String registrationYear;

  Association({
    this.id,
    required this.name,
    required this.phone,
    required this.email,
    required this.address,
    required this.number,
    required this.registrationYear,
  });

  factory Association.fromApi(Map<String, dynamic> json) {
    return Association(
      id: json['_id']?.toString(),
      name: json['name']?.toString() ?? '',
      phone: json['phone']?.toString() ?? '',
      email: json['email']?.toString() ?? '',
      address: json['address']?.toString() ?? '',
      number: json['number']?.toString() ?? '',
      registrationYear: json['registrationYear']?.toString() ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'name': name,
      'phone': phone,
      'email': email,
      'address': address,
      'number': number,
      'registrationYear': registrationYear,
    };
  }

  Association copyWith({
    String? id,
    String? name,
    String? phone,
    String? email,
    String? address,
    String? number,
    String? registrationYear,
  }) {
    return Association(
      id: id ?? this.id,
      name: name ?? this.name,
      phone: phone ?? this.phone,
      email: email ?? this.email,
      address: address ?? this.address,
      number: number ?? this.number,
      registrationYear: registrationYear ?? this.registrationYear,
    );
  }
}

File: backend/models/Association.js
const mongoose = require('mongoose');

const associationSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        trim: true
    },
    number: {
        type: String,
        required: true,
        trim: true
    },
    phone: {
        type: String,
        required: true,
        trim: true
    },
    email: {
        type: String,
        required: true,
        trim: true,
        lowercase: true
    },
    address: {
        type: String,
        required: true,
        trim: true
    },
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
      },
}, {
    timestamps: true
});

const Association = mongoose.model('Association', associationSchema);

module.exports = Association; 